src/cli.rs

```diff
@@ -767,6 +765,22 @@ mod tests {
         assert!(cli.request_items.is_empty());
     }
 
+    #[test]
+    fn method_edge_cases() {
+        // "localhost" is interpreted as method; this is undesirable, but expected
+        parse(&["localhost"]).unwrap_err();
```

Comment:
I was about to suggest adding a special case for `localhost`,  but then I remembered that custom domains defined in `/etc/hosts` would also be affected.